Automattic/newspack-plugin
# add/wizard

### All Submissions:






### Changes proposed in this Pull Request:



Adds a `Wizard` component, which supplants `withWizard` & `withWizardScreen` HOCs. The new component features a more friendly API and state management via `@wordpress/data` (instead of top-level-component state).

For starters, only Reader Revenue wizard is updated to use the new component.

This PR also removes the `WizardPagination` component. It has become effectively the same as `TabbedNavigation` – the only difference being the nav items disabling, added to `TabbedNavigation` via `disableUpcoming` prop.

### How to test the changes in this Pull Request:

This is a pure refactor, everything should work as before.

1. Setup wizard – observe the navigation component (which has changed under the hood) behaves as before
2. Reader Revenue wizard – smoke test.  
3. Other wizards should not be affected in code changes

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201289809401189
